conan/internal/rest/pkg_sign.py

```diff
@@ -12,34 +15,53 @@ def __init__(self, cache, home_folder):
         signer = HomePaths(home_folder).sign_plugin_path
         if os.path.isfile(signer):
             mod, _ = load_python_file(signer)
-            # TODO: At the moment it requires both methods sign and verify, but that might be relaxed
-            self._plugin_sign_function = mod.sign
-            self._plugin_verify_function = mod.verify
+            self._plugin_sign_function = getattr(mod, "sign", None)
+            self._plugin_verify_function = getattr(mod, "verify", None)
         else:
             self._plugin_sign_function = self._plugin_verify_function = None
 
+    @property
+    def is_sign_configured(self):
+        return self._plugin_sign_function is not None
+
+    @property
+    def is_verify_configured(self):
+        return self._plugin_verify_function is not None
+
+    def sign_pkg(self, ref, files, folder):
+        metadata_sign = os.path.join(folder, METADATA, "sign")
+        mkdir(metadata_sign)
+        # Generate the package sign manifest before calling the plugin
+        _save_manifest(folder, metadata_sign)
+        signatures = self._plugin_sign_function(ref, artifacts_folder=folder,
+                                                signature_folder=metadata_sign)
+        # Save signatures file with the plugin's returned signatures data
+        _save_signatures(metadata_sign, signatures)
+        # Add files to package bundle so they get uploaded
+        files[f"{METADATA}/sign/{PKGSIGN_MANIFEST}"] = get_manifest_filepath(metadata_sign)
+        files[f"{METADATA}/sign/{PKGSIGN_SIGNATURES}"] = get_signatures_filepath(metadata_sign)
+        for sig in signatures:
+            for name, file in sig.get("sign_artifacts", {}).items():
+                #TODO: print output?
+                files[f"{METADATA}/sign/{file}"] = os.path.join(metadata_sign, file)
+
     def sign(self, upload_data):
-        if self._plugin_sign_function is None:
+        if not self.is_sign_configured:
```

Comment:
This is never happening, because ``if not pkg_signer.is_sign_configured:`` is checked in the public API before calling this internal signing.
Here, it should be an ``assert`` instead (or just remove it), as this is not intended as regular execution flow.